### PR TITLE
[risk=low][RW-13725] Fix app usage reporting pipeline

### PR DESCRIPTION
**Pull request by Qi77Qi**
Tested locally with smaller batch size 100

```
[Mon Nov 18 13:55:00 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into user table: 2 in 102ms (19.607843 rows/sec)

[Mon Nov 18 13:55:03 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into leonardo_app_usage table: 100 in 150ms (666.666667 rows/sec)

[Mon Nov 18 13:55:04 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into leonardo_app_usage table: 100 in 98ms (1020.408163 rows/sec)

[Mon Nov 18 13:55:05 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into leonardo_app_usage table: 100 in 97ms (1030.927835 rows/sec)

[Mon Nov 18 13:55:07 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into leonardo_app_usage table: 100 in 106ms (943.396226 rows/sec)

[Mon Nov 18 13:55:08 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingUploadServiceImpl uploadBatchTable - Streaming upload into leonardo_app_usage table: 24 in 103ms (233.009709 rows/sec)

[Mon Nov 18 13:55:16 EST 2024] INFO: org.pmiops.workbench.reporting.ReportingVerificationServiceImpl verifyBatchesAndLog - Verifying batches at 1731956099858:
Table   Source  Destination     Difference(%)
workspace       3       3       0 (0.000%)
user    2       2       0 (0.000%)
cohort  0       0       0
leonardo_app_usage      424     424     0 (0.000%)
new_user_satisfaction_survey    0       0       0
user_general_discovery_source   0       0       0
user_partner_discovery_source   0       0       0
```

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java [hunk 1]
```diff
@@ -148,5 +148,9 @@ default <T> Stream<List<T>> getBatchedStream(BiFunction<Long, Long, List<T>> get
 
   int getWorkspaceCount();
 
-  List<ReportingLeonardoAppUsage> getLeonardoAppUsage();
+  List<ReportingLeonardoAppUsage> getLeonardoAppUsage(long limit, long offset);
+
+  default Stream<List<ReportingLeonardoAppUsage>> getBatchedLeonardoAppUsageStream() {
+    return getBatchedStream(this::getLeonardoAppUsage);
+  }
 }
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java:156]: all existing streaming is streamqing query from mysql while app usage is BigQuery. 
Does BQ support this?

Comment (Qi77Qi) [reply api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java:156]: I don't see why it wouldn't as long as `getLeonardoAppUsage` returns the same type

**Review by yonghaoy — COMMENTED** [review]
found a new issue actually. The limit is 10k 
https://github.com/all-of-us/workbench/blob/main/api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java#L70
but the total result in terra side is just 4k for now. We might want a much smaller value for app usage case here like 1k?

**Comment by Qi77Qi** [comment]
@yonghaoy it's using min of the two values, which will be 500 in prod as well

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**